specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-08-01-preview/inference.json

```diff
@@ -0,0 +1,12105 @@
+            ],
+            "default": null
+          },
+          "max_tokens": {
+            "description": "The maximum number of tokens allowed for the generated answer. By default, the number of tokens the model can return will be (4096 - prompt tokens).",
+            "type": "integer",
+            "default": 4096
+          },
+          "presence_penalty": {
+            "description": "Number between -2.0 and 2.0. Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.",
+            "type": "number",
+            "default": 0,
+            "minimum": -2,
+            "maximum": 2
+          },
+          "frequency_penalty": {
+            "description": "Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.",
+            "type": "number",
+            "default": 0,
+            "minimum": -2,
+            "maximum": 2
+          },
+          "logit_bias": {
+            "description": "Modify the likelihood of specified tokens appearing in the completion. Accepts a json object that maps tokens (specified by their token ID in the tokenizer) to an associated bias value from -100 to 100. Mathematically, the bias is added to the logits generated by the model prior to sampling. The exact effect will vary per model, but values between -1 and 1 should decrease or increase likelihood of selection; values like -100 or 100 should result in a ban or exclusive selection of the relevant token.",
+            "type": "object",
+            "nullable": true
+          },
+          "user": {
+            "description": "A unique identifier representing your end-user, which can help Azure OpenAI to monitor and detect abuse.",
+            "type": "string",
+            "example": "user-1234",
+            "nullable": false
+          }
+        }
+      },
+      "createCompletionRequest": {
+        "type": "object",
+        "properties": {
+          "prompt": {
+            "description": "The prompt(s) to generate completions for, encoded as a string, array of strings, array of tokens, or array of token arrays.\n\nNote that <|endoftext|> is the document separator that the model sees during training, so if a prompt is not specified the model will generate as if from the beginning of a new document.\n",
+            "default": "<|endoftext|>",
+            "nullable": true,
+            "oneOf": [
+              {
+                "type": "string",
+                "default": "",
+                "example": "This is a test."
+              },
+              {
+                "type": "array",
+                "items": {
+                  "type": "string",
+                  "default": "",
+                  "example": "This is a test."
+                }
+              },
+              {
+                "type": "array",
```

Comment:
Did we add support for array of tokens for the (legacy) completions request? I thought AOAI never added that...

Reply:
I just tested that we don't support this scenario. I have removed this part in the swagger.